python/kfserving/kfserving/api/kf_serving_client.py

```diff
@@ -289,12 +289,18 @@ def promote(self, name, namespace=None, watch=False, timeout_seconds=600): # pyl
         except KeyError:
             raise RuntimeError("Cannot promote a InferenceService that has no Canary Spec.")
 
+        try:
+            annotations = current_isvc['metadata']['annotations']
+        except KeyError:
+            annotations = dict()
```

Comment:
@jinchihe do you still recall why for promote we need to do replace instead of patch here?

Reply:
@yuzisun we must call `replace`, due to we set `canary=None` here, if we use `patch`, the canary will not be removed, and keep old one. To resolve this issue, we implemented `replace` API ...